ramanujan/pcf/pcf.py

```diff
@@ -155,25 +158,25 @@ def limit(self, depth, start=1, vector=ramanujan.zero()) -> ramanujan.Matrix:
         Returns:
             The pcf convergence limit as defined above.
         """
-        return self.walk(depth, start) * vector
+        return (self.walk(depth, start)[-1]) * vector
 
     def delta(self, depth, limit=None):
         r"""
         Calculates the irrationality measure $\delta$ defined, as:
         $|\frac{p_n}{q_n} - L| = \frac{1}{q_n}^{1+\delta}$
 
-        If `limit` is not specified (i.e, `limit is None`),
-        then `limit` is approximated as `limit = self.limit(2 * depth)`
+        If limit is not specified (i.e, limit is None),
+        then limit is approximated as limit = self.limit(2 * depth)
 
         Args:
             depth: $n$
             limit: $L$
         Returns:
             the delta value as defined above.
         """
-        m = self.walk(depth)
+        m = self.walk(depth)[-1]
         p, q = m * ramanujan.zero()
         if limit is None:
-            m *= self.M().walk({n: 1}, depth, {n: depth})
+            m *= (self.M().walk({n: 1}, depth, {n: depth}))[-1]
```

Comment:
Since you implemented the logic of returning a list, why not use it?
Check if limit is `None`, if it is then get results for `depth, 2*depth`.